File: src/api/metadata/sports.ts
import type { HttpClient } from '../../utils/http.js';
import type { SportsFilters } from '../../types/index.js';

/**
 * API for retrieving sports-specific filters.
 *
 * Get available filters for sports-related prediction markets
 * (leagues, teams, event types, etc.).
 *
 * @example
 * ```typescript
 * const dflow = new DFlowClient();
 *
 * const filters = await dflow.sports.getFiltersBySports();
 * console.log(filters.leagues);
 * ```
 */
export class SportsAPI {
  constructor(private http: HttpClient) {}

  /**
   * Get all available sports filters.
   *
   * @returns Sports filters including leagues, teams, and event types
   *
   * @example
   * ```typescript
   * const filters = await dflow.sports.getFiltersBySports();
   * filters.leagues.forEach(league => console.log(league.name));
   * ```
   */
  async getFiltersBySports(): Promise<SportsFilters> {
    return this.http.get<SportsFilters>('/filters_by_sports');
  }
}

File: python/src/dflow/api/metadata/sports.py
"""Sports API for DFlow SDK."""

from dflow.types import SportsFilters
from dflow.utils.http import HttpClient


class SportsAPI:
    """API for retrieving sports-specific filters.

    Get available filters for sports-related prediction markets
    (leagues, teams, event types, etc.).

    Example:
        >>> dflow = DFlowClient()
        >>> filters = dflow.sports.get_filters_by_sports()
        >>> for sport in filters.sports:
        ...     print(f"{sport.sport}: {sport.leagues}")
    """

    def __init__(self, http: HttpClient):
        self._http = http

    def get_filters_by_sports(self) -> SportsFilters:
        """Get all available sports filters.

        Returns:
            Sports filters including leagues, teams, and event types

        Example:
            >>> filters = dflow.sports.get_filters_by_sports()
            >>> for sport in filters.sports:
            ...     print(f"{sport.sport}: {sport.leagues}")
        """
        data = self._http.get("/filters_by_sports")
        return SportsFilters.model_validate(data)
